### DISCUSSION TITLE
Transposing matrices

### DISCUSSION THREAD

**Discussion by ansost** [RePsychLing/SMLP2023#44] [Q&A] [answered]
I know I can wrap `copy` or `Matrix` around `transpose` but is there a function that transposes a matrix and returns a matrix (not just a view)?

Im sure there is an easy answer to this I just can't find it:D

**Comment by palday** [comment, marked as answer]
nope, you need the explicit call to `copy`. :) It's a feature, not a bug. :wink:

**Comment by dmbates** [comment]
A more canonical way of instantiating an `Array` is with `collect`.  It is guaranteed to allocate and populate an `Array` from some kind of expression which may involve delayed evaluation.
